src/Manager.php

```diff
@@ -52,100 +50,65 @@ public function userHasPermission(
             return $this->guestHasPermission($permissionName);
         }
 
-        $userId = (string) $userId;
-        $assignments = $this->assignmentsStorage->getByUserId($userId);
+        $permission = $this->itemsStorage->getPermission($permissionName);
+        if ($permission === null) {
+            return false;
+        }
 
-        if (empty($assignments)) {
+        $userId = (string) $userId;
+        if (!$this->executeRule($userId, $permission, $parameters)) {
             return false;
         }
 
-        return $this->userHasItem(
-            $userId,
-            $this->itemsStorage->getPermission($permissionName),
-            $parameters,
-            $assignments
-        );
+        $parentPermissions = $this->itemsStorage->getParents($permission->getName());
+        $permissionNames = [$permission->getName(), ...array_keys($parentPermissions)];
+
+        return $this->assignmentsStorage->userHasItem($userId, $permissionNames);
```

Comment:
Seems, this methods does not account that parent permissions also can contain rules, right?

Reply:
I think this how it was before these changes too and suggest to solve it separately.

Reply:
#193